src/Particle/Par_SetParUID.cpp

```diff
@@ -0,0 +1,144 @@
+#include "GAMER.h"
+
+
+
+#ifdef PARTICLE
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Par_SetParUID
+// Description :  Set the particle unique UID
+//
+// Note        :  1. This function should be done before initializaion
+//                2. Invoked by Init_GAMER() and SF_CreateStar()
+//                3. Assuming all the particles is active when initializing
+//                4. The new particle UID should be initialized as -1
+//                5. Currently, the new particles UID are given by the sorted position
+//
+// Paramter    :  init : Initialization stage or not
```

Comment:
I am wondering what the reason is to distinguish whether it is the initialization stage or not.
It seems to me that the non-initial mode is a more general method, and it can be applied to the initial mode.

And although in the current usage of `Par_Init_ByFunction_*()` the particle data are all constructed and scattered out by Rank-0, it seems to me there is no restriction for each MPI rank to construct its own particles in there (see Note 3.), and the order of particles among ranks may depend on the number of ranks and differ from the orders here.
Therefore, I think it will be safer to sort the particles by position before assigning their UID, as it is done in the non-initial mode.